storage/integration_test.go

```diff
@@ -2797,24 +2796,20 @@ func TestIntegration_RequesterPays(t *testing.T) {
 		return -1
 	}
 
-	bucketName := uidSpace.New()
-	requesterPaysBucket := mainUserClient.Bucket(bucketName)
-	bucketName2 := uidSpace.New()
-	requesterPaysBucket2 := mainUserClient.Bucket(bucketName2)
+	// Retry requests if idempotent or if it is a rate limit issue
+	retryRateLimit := func(err error) bool {
+		var gErr *googleapi.Error
 
-	// Create a requester-pays bucket. The bucket is contained in the project mainProjectID
-	h.mustCreate(requesterPaysBucket, mainProjectID, &BucketAttrs{RequesterPays: true})
-	if err := requesterPaysBucket.ACL().Set(ctx, ACLEntity("user-"+otherUserEmail), RoleOwner); err != nil {
-		t.Fatalf("set ACL: %v", err)
-	}
-	defer h.mustDeleteBucket(requesterPaysBucket)
+		isRateLimitErr := false
+		if xerrors.As(err, &gErr) {
+			isRateLimitErr = gErr.Code == 429
+		}
 
-	// Create a second requester-pays bucket to avoid rate limits.
-	h.mustCreate(requesterPaysBucket2, mainProjectID, &BucketAttrs{RequesterPays: true})
-	if err := requesterPaysBucket2.ACL().Set(ctx, ACLEntity("user-"+otherUserEmail), RoleOwner); err != nil {
-		t.Fatalf("set ACL: %v", err)
+		return isRateLimitErr || shouldRetry(err)
 	}
-	defer h.mustDeleteBucket(requesterPaysBucket2)
+
+	mainUserClient.SetRetry(WithErrorFunc(retryRateLimit))
```

Comment:
I think instead we can probably just make the policy RetryAlways for these tests? No real idempotency concerns in a test.